hbase-common/src/main/java/org/apache/hadoop/hbase/ByteBufferTag.java

```diff
@@ -44,6 +44,21 @@ public ByteBufferTag(ByteBuffer buffer, int offset, int length) {
     this.type = ByteBufferUtils.toByte(buffer, offset + TAG_LENGTH_SIZE);
   }
 
+  /**
+   * This constructor <b>just contains the tag value</b> in ByteBuffer. In above constructor,
+   * tag length and tag type are also contained in the ByteBuffer.
+   * @param buffer input byte buffer
+   * @param type  type of the tag
+   * @param offset offset of the start of tag in byte buffer
+   * @param length length of the tag
+   */
+  public ByteBufferTag(ByteBuffer buffer, byte type, int offset, int length) {
```

Comment:
So here Tag value is in the BB at given offset -> length.
Suggest rearrange the params order so that BB, offset and length are together.
May be byte type, BB value, int offset, int length) will be better
Also pls name BB param as value  than plain buffer.

Reply:
Thank you for the review. Addressed it in latest commit.